### PR TITLE
feature(connections): disable ConnectionInfo table editing when client was connected

### PR DESCRIPTION
**Pull request by oceanlvr**
The following is the previous version:

before

We expect that the form cannot be modified when connected, Now we cannot modify the form while connected.

after

### CODE REVIEW

File: src/views/connections/ConnectionInfo.vue [hunk 1]
```diff
@@ -139,6 +151,11 @@ export default class ConnectionInfo extends Vue {
 
   // Reverse the status of clientIdWithTime.
   private reverseClientIDWithTime() {
+    // when the client connected, we should disable this icon event
+    let curClientConnection: any = this.client
```
Comment (ysfscream) [inline src/views/connections/ConnectionInfo.vue:155]: `this.client` is not the `any` type.

File: src/views/connections/ConnectionInfo.vue [hunk 2]
```diff
@@ -179,6 +196,11 @@ export default class ConnectionInfo extends Vue {
   }
 
   private refreshClientId() {
+    // when the client connected, we should disable this icon event
+    let curClientConnection: any = this.client
+    if (curClientConnection.connected === true) {
+      return
+    }
```
Comment (ysfscream) [inline src/views/connections/ConnectionInfo.vue:203]: The code here is duplicated, please consider modifying it as a function.

File: src/views/connections/ConnectionInfo.vue [hunk 3]
```diff
@@ -139,6 +151,11 @@ export default class ConnectionInfo extends Vue {
 
   // Reverse the status of clientIdWithTime.
   private reverseClientIDWithTime() {
+    // when the client connected, we should disable this icon event
+    let curClientConnection: any = this.client
+    if (curClientConnection.connected === true) {
```
Comment (ysfscream) [inline src/views/connections/ConnectionInfo.vue:156]:
```js
if (curClientConnection.connected) {}
```

It will be easier to write like this.

**Comment by ysfscream** [comment]
> We expect that the form cannot be modified when connected, Now we cannot modify the form while connected.
> after

Disabled input seems a bit ugly. Maybe you can set the `input` background color to `transparent`, and see some. Then look at the effect and try not to destroy the original style, our purpose is just to disable it.

File: src/views/connections/ConnectionInfo.vue [hunk 4]
```diff
@@ -26,28 +26,40 @@
...
               </a>
             </el-tooltip>
-            <el-input size="mini" v-model="connection.clientId">
-              <i slot="suffix" title="Refresh" class="el-icon-refresh" @click="refreshClientId"> </i>
+            <el-input size="mini" v-model="connection.clientId" :disabled="client.connected">
+              <i
+                slot="suffix"
+                title="Refresh"
+                class="el-icon-refresh"
+                :class="client.connected === true ? 'icon-oper-disable' : ''"
```
Comment (0721Betty) [inline src/views/connections/ConnectionInfo.vue:34]: client.connected?

Comment (oceanlvr) [reply src/views/connections/ConnectionInfo.vue:34]: Sorry, I will pay attention to this detail.

**Comment by oceanlvr** [comment]
before

image

after

image

**Review by 0721Betty — APPROVED** [review]

**Review by ysfscream — APPROVED** [review]

**Pull request merged**